hooks/tk-multi-publish2/basic/publish_mari_session.py

```diff
@@ -0,0 +1,419 @@
+        Shotgun will also reflect the version number identified in the filename.
+        The basic worklfow recognizes the following version formats by default:
+
+        <ul>
+        <li><code>filename.v###.ext</code></li>
+        <li><code>filename_v###.ext</code></li>
+        <li><code>filename-v###.ext</code></li>
+        </ul>
+
+        After publishing, if a version number is detected in the file, the file
+        will automatically be saved to the next incremental version number.
+        For example, <code>filename.v001.ext</code> will be published and copied
+        to <code>filename.v002.ext</code>
+
+        If the next incremental version of the file already exists on disk, the
+        validation step will produce a warning, and a button will be provided in
+        the logging output which will allow saving the session to the next
+        available version number prior to publishing.
+
+        <br><br><i>NOTE: any amount of version number padding is supported.</i>
+
+        <h3>Overwriting an existing publish</h3>
+        A file can be published multiple times however only the most recent
+        publish will be available to other users. Warnings will be provided
+        during validation if there are previous publishes.
+        """ % (loader_url,)
+        # TODO: add link to workflow docs
+
+    @property
+    def settings(self):
+        """
+        Dictionary defining the settings that this plugin expects to receive
+        through the settings parameter in the accept, validate, publish and
+        finalize methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
+        part of its environment configuration.
+        """
+        return {
+            "Publish Type": {
+                "type": "shotgun_publish_type",
+                "default": "UDIM Image",
+                "description": "SG publish type to associate publishes with."
+            },
+            "Publish Template": {
+                "type": "template",
+                "default": None,
+                "description": "Template path for published work files. Should"
+                               "correspond to a template defined in "
+                               "templates.yml.",
+            }
+        }
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["maya.*", "file.maya"]
+        """
+        return ["mari.texture"]

```

Comment:
So the plugin is a "session" publish but operating on textures? Maybe we should rename or clarify?